File: 剑指offer/22链表中倒数第k个节点.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func getKthFromEnd(head *ListNode, k int) *ListNode {
    dummy := &ListNode{Val:0, Next: head}
    fast := dummy
    for i := 0; i < k && fast != nil; i++ {
        fast = fast.Next
    }

    slow := dummy
    for fast != nil {
        slow = slow.Next
        fast = fast.Next
    }
    return slow
}

File: 剑指offer/22链表中倒数第k个节点.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getKthFromEnd(self, head: ListNode, k: int) -> ListNode:
        dummy = ListNode(0, head)
        slow, fast = dummy, dummy
        while k > 0 and fast:
            fast = fast.next
            k -= 1

        while fast:
            slow = slow.next
            fast = fast.next

        return slow


